R/join_ald_scenario.R

```diff
@@ -38,101 +27,35 @@
                               region_isos = r2dii.data::region_isos,
                               add_green_technologies = FALSE) {
-  check_portfolio_ald_scenario(data, ald, scenario)
-
-  # Track provenance to avoid clash in the column name "source"
-  region_isos <- region_isos %>%
-    rename(scenario_source = .data$source)
-
-  ald <- modify_at_(ald, "sector", tolower)
-  ald <- modify_at_(ald, "technology", tolower)
-
-  if (add_green_technologies) {
-    ald <- add_green_technologies_to_ald(ald, scenario)
+  lifecycle::deprecate_warn(
+    "0.2.0",
+    "join_ald_scenario()",
+    "join_abcd_scenario()"
+    )
+
+  if ("name_ald" %in% names(data)) {
+    data <- rename(data, name_abcd = .data$name_ald)
   }
 
-  out <- data %>%
-    left_join(ald, by = ald_columns()) %>%
-    inner_join(scenario, by = scenario_columns()) %>%
-    warn_if_has_zero_rows("Joining `scenario` outputs 0 rows.") %>%
-    mutate(plant_location = tolower(.data$plant_location)) %>%
-    inner_join(
-      region_isos,
-      by = c("region", "plant_location" = "isos", "scenario_source")
-    ) %>%
-    warn_if_has_zero_rows("Joining `region_isos` outputs 0 rows.")
-  out
-}
-
-warn_if_has_zero_rows <- function(data, message) {
-  if (nrow(data) == 0L) warn(message = message, class = "has_zero_rows")
-  invisible(data)
-}
-
-check_portfolio_ald_scenario <- function(valid_matches, ald, scenario) {
-  check_crucial_names(valid_matches, names(ald_columns()))
-  walk_(names(ald_columns()), ~ check_no_value_is_missing(valid_matches, .x))
-
-  check_crucial_names(
-    ald, c("name_company", "plant_location", unname(scenario_columns()))
-  )
-  walk_(
-    c("name_company", unname(scenario_columns())),
-    ~ check_no_value_is_missing(ald, .x)
-  )
-
-
-  check_crucial_names(scenario, c(scenario_columns(), "scenario_source", "region"))
-  walk_(
-    c(scenario_columns(), "scenario_source", "region"),
-    ~ check_no_value_is_missing(scenario, .x)
-  )
-
-  invisible(valid_matches)
-}
-
-add_green_technologies_to_ald <- function(data, scenario) {
-  green_techs <- r2dii.data::green_or_brown %>%
-    filter(.data$green_or_brown == "green") %>%
-    select(-.data$green_or_brown)
-
-  green_techs_in_scenario <- scenario %>%
-    select(.data$sector, .data$technology) %>%
-    unique() %>%
-    inner_join(green_techs, by = c("sector", "technology"))
-
-  green_rows_to_add <- data %>%
-    group_by(
-      .data$name_company,
-      .data$sector,
-      .data$year,
-      .data$plant_location,
-      .data$is_ultimate_owner
-    ) %>%
-    summarize() %>%
-    left_join(green_techs_in_scenario, by = "sector") %>%
-    mutate(production = 0)
-
-  dplyr::bind_rows(data, green_rows_to_add)
-}
+  if ("sector_ald" %in% names(data)) {
+    data <- rename(data, sector_abcd = .data$sector_ald)
+  }
 
-ald_columns <- function() {
-  c(
-    name_ald = "name_company",
-    sector_ald = "sector"
-  )
-}
+  join_abcd_scenario(data, ald, scenario, region_isos, add_green_technologies)
```

Comment:
deprecate message, change column names, use `join_abcd_scenario()` instead... got it 👍🏻

Reply:
eaxctly.